porch/pkg/engine/engine.go

```diff
@@ -104,10 +112,56 @@ func (cad *cadEngine) ListPackageRevisions(ctx context.Context, repositorySpec *
 	return repo.ListPackageRevisions(ctx, filter)
 }
 
-func (cad *cadEngine) CreatePackageRevision(ctx context.Context, repositoryObj *configapi.Repository, obj *api.PackageRevision) (repository.PackageRevision, error) {
+func buildPackageConfig(ctx context.Context, obj *api.PackageRevision, parent repository.PackageRevision) (*builtins.PackageConfig, error) {
```

Comment:
Can you remind me.. what `parent` means in context of porch ? few things popping up in my mind: `upstream` package, `parent` of a draft package and parent package of a subpackage (which we don't support in porch yet).

Reply:
Not sure I've got the best name, but `parent` will let us establish a "position" of the package, in the same way that the directory structure does in git.

This proves to be a useful concept, I think, for value propagation and just to organize things.

This isn't done yet, but I would hope we could map this to directories in a git repo, so we would likely need this for CLI/server parity.

Reply:
`positionalParent` is how I will remember this now :)